### PR TITLE
feat(chip): Stage 3 Intents and Content

### PR DESCRIPTION
**Pull request by andresin87**
## feat(chip): intents

**TASK**: 

### Types of changes

- [x] ✨ New feature (non-breaking change which adds functionality)

### Screenshots - Animations

ADDS Content and ClearButton

### CODE REVIEW

**Comment by andresin87** [comment]
i have to replace the forced values by the them calculed fns

File: packages/components/chip/src/variants/dashed.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
+import { tw } from '@spark-ui/internal-utils'
+
+export const dashedVariants = [
+  {
+    intent: 'primary',
+    design: 'dashed',
+    class: tw([
+      'enabled:hover:bg-primary/dim-5',
```
Comment (acd02) [inline packages/components/chip/src/variants/dashed.ts:8]: 💅 , just a reminder that when using the `enabled` pseudo-class, these styles won't be applied if someone were to use the `asChild` pattern with an element that is not a button

**Review by acd02 — APPROVED** [review]

File: packages/components/chip/src/variants/dashed.ts (hunk 1 above)
Comment (andresin87) [reply packages/components/chip/src/variants/dashed.ts:8]: for sure, and if they don't add an onClick prop the default element is a button and this styles are never applied, (by default)

File: packages/components/chip/src/ChipContent.styles.tsx [hunk 2]
```diff
@@ -0,0 +1,5 @@
+import { cva, type VariantProps } from 'class-variance-authority'
+
+export const chipContentStyles = cva(['inline-block grow truncate'])
+
+export type ChipContentStylesProps = VariantProps<typeof chipContentStyles>
```
Comment (Powerplex) [inline packages/components/chip/src/ChipContent.styles.tsx:5]: I think you forgot to delete this file.

**Comment by andresin87** [comment]
@Powerplex @kikoruiz @soykje @andresz1

File: packages/components/chip/src/Chip.stories.tsx [hunk 3]
```diff
@@ -7,11 +8,60 @@ const meta: Meta<typeof Chip> = {
...
+            design={design}
+            key={`${design}-${intent}`}
+            intent={intent}
+            onClick={() => console.log(`click ${design} ${intent}`)}
+          >
+            {intent} chip huge huge chip long and wrong
+          </Chip>
+        ))}
+      </div>
+    ))}
```
Comment (andresz1) [inline packages/components/chip/src/Chip.stories.tsx:65]: WDTY is we only display the intent here ? the `Action` could be another story

File: packages/components/chip/src/Chip.stories.tsx [hunk 4]
```diff
@@ -7,11 +8,60 @@ const meta: Meta<typeof Chip> = {
...
-    Hello World!
-    <Chip.ClearButton onClick={() => console.log('clear')} />
-  </Chip>
+  <div className="flex flex-col items-start gap-md">
+    {designs.map(design => (
+      <Chip design={design} key={design}>
+        <Chip.Content>Hello World!</Chip.Content>
+        <Chip.ClearButton onClick={() => console.log(`clear ${design}`)} />
+      </Chip>
+    ))}
```
Comment (andresz1) [inline packages/components/chip/src/Chip.stories.tsx:35]: Perhaps we need a `design` story ? and a clear button story ?

File: packages/components/chip/src/ChipContent.tsx [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import React, { ComponentPropsWithoutRef, forwardRef } from 'react'
+
+export const ChipContent = forwardRef<HTMLSpanElement, ComponentPropsWithoutRef<'span'>>(
```
Comment (andresz1) [inline packages/components/chip/src/ChipContent.tsx:3]: Can you export the `ChipContentProps` ?

File: packages/components/chip/src/Chip.tsx [hunk 6]
```diff
@@ -75,7 +77,12 @@ export const Chip = forwardRef<HTMLButtonElement | HTMLDivElement, ChipProps>(
       disabled: !!disabled,
     })
 
-    const hasClearButton = findElement('Chip.ClearButton')(children)
+    const findChipElement = findElement(children)
+
+    const hasClearButton = findChipElement('Chip.ClearButton')
+    const hasContent = findChipElement('Chip.Content')
+
+    const ChildrenWrapper = hasContent ? Fragment : ChipContent
```
Comment (andresz1) [inline packages/components/chip/src/Chip.tsx:85]:
```suggestion
    const Content = hasContent ? Fragment : ChipContent
```

**Review by andresz1 — APPROVED** [review]

File: packages/components/chip/src/Chip.stories.tsx (hunk 4 above)
Comment (andresin87) [reply packages/components/chip/src/Chip.stories.tsx:35]: stories Pr are not created yet. This is just to test it working

File: packages/components/chip/src/Chip.stories.tsx (hunk 3 above)
Comment (andresin87) [reply packages/components/chip/src/Chip.stories.tsx:65]: for sure. This is a porototype for testing it working on both usages. The PR for stories is not ready yet and will cover all the required properties variations

File: packages/components/chip/src/ChipContent.tsx (hunk 5 above)
Comment (andresin87) [reply packages/components/chip/src/ChipContent.tsx]: done

File: packages/components/chip/src/Chip.tsx (hunk 6 above)
Comment (andresin87) [reply packages/components/chip/src/Chip.tsx]: done

**Pull request merged**